packages/block-library/src/query/utils.js

```diff
@@ -136,11 +136,15 @@ export const usePostTypes = () => {
 export const useTaxonomies = ( postType ) => {
 	const taxonomies = useSelect(
 		( select ) => {
-			const { getTaxonomies } = select( coreStore );
-			return getTaxonomies( {
-				type: postType,
-				per_page: -1,
-			} );
+			const { getTaxonomies, getPostType } = select( coreStore );
+			// Does the post type have taxonomies?
+			if ( getPostType( postType )?.taxonomies?.length > 0 ) {
```

Comment:
This logic should be inside getTaxonomies whenever `type` is provided. If only I could find that dynamically created function, I could state exactly where and explain why.

Reply:
The `getEntityRecords` (the `getTaxonomies` is just a shortcut) is a simple wrapper around `apiFetch`; it's not and shouldn't be aware of relations between entities.

Think of it as `get_object_taxonomies( 'page' )`; running it will return an empty array, but REST API returns an error because that's how APIs communicate.

It's up to code consumers to guard against similar scenarios.

Reply:
If the API fails at its job, it should be fixed upstream. This is just patching and it'll just turn into code debt. The disconnect between Core and Gutenberg, and the gaslighting that happens in Slack, is disconcerting to say the least.

Reply:
@ntsekouras, that is how REST API communicates; instead of failing silently and returning an empty array like `get_object_taxonomies( 'page' )`, it returns an error so you can update your queries.